src/main/java/org/jabref/logic/integrity/DoiDuplicationChecker.java

```diff
@@ -0,0 +1,51 @@
+package org.jabref.logic.integrity;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import javafx.collections.ObservableList;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.identifier.DOI;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+public class DoiDuplicationChecker implements Checker {
+    private final BibDatabase database;
+    private Map<BibEntry, List<IntegrityMessage>> errors;
+
+    public DoiDuplicationChecker(BibDatabase database) {
+        this.database = Objects.requireNonNull(database);
+    }
+
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        if (errors == null) {
+            errors = new HashMap<>();
+
+            ObservableList<BibEntry> bibEntries = database.getEntries();
+            BiMap<DOI, List<BibEntry>> duplicateMap = HashBiMap.create(bibEntries.size());
+            for (BibEntry bibEntry : bibEntries) {
+                bibEntry.getDOI().ifPresent(doi ->
+                        duplicateMap.computeIfAbsent(doi, x -> new ArrayList<>()).add(bibEntry));
```

Comment:
```suggestion
                        duplicateMap.computeIfAbsent(doi, entry -> new ArrayList<>()).add(bibEntry));
```

Reply:
`entry` is a variable name passed to the method (the entry to be checked). Thus, the variable name cannot be used in a lambda expression.

Reply:
@koppor Please have a close look. The second argument of duplicateMap.computeIfAbsent refers to the value and this clearly is of type  List<BibEntry>.   So then maybe name it `entries`

Reply:
@Siedlerchr Please explain the JAvaDoc to me. The Map is FROM doi TO list. https://www.geeksforgeeks.org/hashmap-computeifabsent-method-in-java-with-examples/ explains the `computeIfAbsent`.

To me, it reads, that the **key** is passed to `Function<? super K, ? extends V> remappingFunction`. The **key** is clearly (!) a DOI not a list. -- Am I wrong?